src/org/labkey/cds/CDSController.java

```diff
@@ -191,12 +194,21 @@ public Object execute(Object o, BindException errors) throws Exception
             ApiSimpleResponse response = new ApiSimpleResponse();
             List<Item> items = new ArrayList<>();
 
-            List<RSSFeed> feeds = RSSService.get().getFeeds(getContainer(), getUser());
+            User user = getUser();
+
+            // In order to display blog posts on Dataspace's public page where the user is not required to log-in, we need to create a
+            // Limited user with Reader role otherwise no feeds will be returned by RSSService.get().getFeeds()
+            if (getUser().isGuest())
+            {
+                user = new LimitedUser(User.guest, new int[0], Collections.singleton(RoleManager.getRole(ReaderRole.class)), false);
```

Comment:
We should also be able to reuse the current user object versus `User.guest`
```suggestion
                user = new LimitedUser(user, new int[0], Collections.singleton(RoleManager.getRole(ReaderRole.class)), false);
```

Reply:
Good point. Done!